### ISSUE TITLE
memory corruption

### ISSUE THREAD

**Issue by leviswind** [facebookresearch/video-long-term-feature-banks#8]
I'm a freshman of caffe2. When I run the training code,

python2 tools/train_net.py \
  --config_file configs/ava_r50_baseline.yaml \
  CHECKPOINT.DIR /tmp/baseline-output

we meet the following error,

/mnt/cephfs_wj/vc/levi.li/codes/video-long-term-feature-banks/lib/core/config.py:427: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  yaml_config = AttrDict(yaml.load(fopen))
Ignoring @/caffe2/caffe2/contrib/nccl:nccl_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/contrib/gloo:gloo_ops as it is not a valid file.
Ignoring @/caffe2/caffe2/contrib/gloo:gloo_ops_gpu as it is not a valid file.
Found Detectron ops lib: /mnt/cephfs_wj/vc/levi.li/software/anaconda2/lib/python2.7/site-packages/torch/lib/libcaffe2_detectron_ops_gpu.so
[INFO: config.py:  369]: Config:
[INFO: config.py:  370]: {'AVA': {'ANNOTATION_DIR': 'data/ava/annotations',
         'DETECTION_SCORE_THRESH_EVAL': [0.85],
         'DETECTION_SCORE_THRESH_TRAIN': 0.9,
         'FORCE_TEST_FLIP': False,
         'FPS': 30,
         'FRAME_LIST_DIR': 'data/ava/frame_lists',
         'FULL_EVAL_DURING_TRAINING': False,
         'LFB_DETECTION_SCORE_THRESH': 0.9,
         'LFB_MAX_NUM_FEAT_PER_STEP': 5,
         'TEST_BOX_LISTS': ['ava_val_predicted_boxes.csv'],
         'TEST_LFB_BOX_LISTS': ['ava_val_predicted_boxes.csv'],
         'TEST_LISTS': ['val.csv'],
         'TEST_MULTI_CROP': False,
         'TEST_MULTI_CROP_SCALES': [224, 256, 320],
         'TEST_ON_TEST_SET': False,
         'TRAIN_BOX_LISTS': ['ava_train_v2.1.csv',
                             'ava_train_predicted_boxes.csv'],
         'TRAIN_LFB_BOX_LISTS': ['ava_train_predicted_boxes.csv'],
         'TRAIN_LISTS': ['train.csv'],
         'TRAIN_ON_TRAIN_VAL': False},
 'CHARADES': {'FPS': 24,
              'FRAME_LIST_DIR': 'data/charades/frame_lists',
              'LFB_CLIPS_PER_SECOND': 2,
              'NUM_TEST_CLIPS_DURING_TRAINING': 9,
              'NUM_TEST_CLIPS_FINAL_EVAL': 30,
              'TEST_LISTS': ['val.csv'],
              'TRAIN_LISTS': ['train.csv']},
 'CHECKPOINT': {'CHECKPOINT_MODEL': True,
                'CHECKPOINT_PERIOD': 4000,
                'CONVERT_MODEL': True,
                'DIR': '/tmp/baseline-output',
                'RESUME': True},
 'CUDNN_WORKSPACE_LIMIT': 256,
 'DATADIR': 'data/ava/frames',
 'DATALOADER': {'MAX_BAD_IMAGES': 100},
 'DATASET': 'ava',
 'DATA_MEAN': [0.45, 0.45, 0.45],
 'DATA_STD': [0.225, 0.225, 0.225],
 'DEBUG': False,
 'EPIC': {'ANNOTATIONS': 'EPIC_train_action_labels.csv',
          'ANNOTATION_DIR': 'data/epic/annotations',
          'CLASS_TYPE': '',
          'FPS': 30,
          'FRAME_LIST_DIR': 'data/epic/frame_lists',
          'MAX_NUM_FEATS_PER_NOUN_LFB_FRAME': 10,
          'NOUN_LFB_FRAMES_PER_SECOND': 1,
          'TEST_LISTS': ['val.csv'],
          'TRAIN_LISTS': ['train.csv'],
          'VERB_LFB_CLIPS_PER_SECOND': 1},
 'FBO_NL': {'DROPOUT_RATE': 0.2,
            'INPUT_DROPOUT_ON': True,
            'INPUT_REDUCE_DIM': True,
            'LATENT_DIM': 512,
            'LFB_DROPOUT_ON': True,
            'NL_DROPOUT_ON': True,
            'NUM_LAYERS': 2,
            'PRE_ACT': True,
            'PRE_ACT_LN': True,
            'SCALE': True},
 'GET_TRAIN_LFB': False,
 'IMG_LOAD_RETRY': 10,
 'INTERPOLATION': 'INTER_LINEAR',
 'LFB': {'ENABLED': False,
         'FBO_TYPE': 'nl',
         'LFB_DIM': 2048,
         'LOAD_LFB': False,
         'LOAD_LFB_PATH': '',
         'MODEL_PARAMS_FILE': '',
         'NUM_LFB_FEAT': 500,
         'WINDOW_SIZE': 100,
         'WRITE_LFB': False},
 'LOG_PERIOD': 10,
 'MINIBATCH_QUEUE_SIZE': 64,
 'MODEL': {'ALLOW_INPLACE_RELU': True,
           'ALLOW_INPLACE_RESHAPE': True,
           'ALLOW_INPLACE_SUM': True,
           'BN_EPSILON': 1.0000001e-05,
           'BN_INIT_GAMMA': 0.0,
           'BN_MOMENTUM': 0.9,
           'DEPTH': 50,
           'DILATIONS_AFTER_CONV5': True,
           'FC_INIT_STD': 0.01,
           'FREEZE_BACKBONE': False,
           'MEAN': 114.75,
           'MEMONGER': True,
           'MODEL_NAME': 'resnet_video',
           'MULTI_LABEL': True,
           'NUM_CLASSES': 80,
           'SAMPLE_THREADS': 8,
           'STD': 57.375,
           'USE_AFFINE': True,
           'USE_BGR': False,
           'VIDEO_ARC_CHOICE': 2},
 'NONLOCAL': {'BN_EPSILON': 1.0000001e-05,
              'BN_INIT_GAMMA': 0.0,
              'BN_MOMENTUM': 0.9,
              'CONV3_NONLOCAL': True,
              'CONV4_NONLOCAL': True,
              'CONV_INIT_STD': 0.01,
              'LAYER_MOD': 2,
              'NO_BIAS': 0,
              'USE_AFFINE': True,
              'USE_BN': False,
              'USE_MAXPOOL': True,
              'USE_SCALE': True,
              'USE_SOFTMAX': True,
              'USE_ZERO_INIT_CONV': True},
 'NUM_GPUS': 1,
 'PROF_DAG': False,
 'RESNETS': {'NUM_GROUPS': 1,
             'STRIDE_1X1': False,
             'TRANS_FUNC': 'bottleneck_transformation_3d',
             'WIDTH_PER_GROUP': 64},
 'RNG_SEED': 2,
 'ROI': {'SCALE_FACTOR': 16, 'XFORM_RESOLUTION': 7},
 'ROOT_GPU_ID': 0,
 'SOLVER': {'BASE_LR': 0.04,
            'GAMMA': 0.1,
            'LRS': [1, 0.1, 0.01, 0.001],
            'LR_POLICY': 'steps_with_relative_lrs',
            'MAX_ITER': 140000,
            'MOMENTUM': 0.9,
            'NESTEROV': True,
            'SCALE_MOMENTUM': True,
            'SCALE_MOMENTUM_THRESHOLD': 1.1,
            'STEPS': [0, 100000, 120000, 140000],
            'STEP_SIZES': [100000, 20000, 20000],
            'WARMUP': {'WARMUP_END_ITER': 2000,
                       'WARMUP_ON': True,
                       'WARMUP_START_LR': 0.01},
            'WEIGHT_DECAY': 1e-06,
            'WEIGHT_DECAY_BN': 0.0},
 'TEST': {'BATCH_SIZE': 16,
          'CROP_SHIFT': 1,
          'CROP_SIZE': 256,
          'DATASET_SIZE': 64,
          'DATA_TYPE': 'val',
          'PARAMS_FILE': '',
          'SAMPLE_RATE': 2,
          'SCALE': 256,
          'VIDEO_LENGTH': 32},
 'TRAIN': {'BATCH_SIZE': 2,
           'COMPUTE_PRECISE_BN': False,
           'CROP_SIZE': 224,
           'DATASET_SIZE': 235,
           'DATA_TYPE': 'train',
           'DROPOUT_RATE': 0.3,
           'EVAL_PERIOD': 8000,
           'ITER_COMPUTE_PRECISE_BN': 200,
           'JITTER_SCALES': [256, 320],
           'PARAMS_FILE': 'pretrained_weights/r50_k400_pretrained.pkl',
           'PCA_EIGVAL': [0.225, 0.224, 0.229],
           'PCA_EIGVEC': [[-0.5675, 0.7192, 0.4009],
                          [-0.5808, -0.0045, -0.814],
                          [-0.5836, -0.6948, 0.4203]],
           'RESET_START_ITER': True,
           'RESUME_FROM_BATCH_SIZE': -1,
           'SAMPLE_RATE': 2,
           'TEST_AFTER_TRAIN': True,
           'USE_COLOR_AUGMENTATION': False,
           'VIDEO_LENGTH': 32},
 'USE_CYTHON': False}
[E init_intrinsics_check.cc:43] CPU feature avx is present on your machine, but the Caffe2 binary is not compiled with it. It means you may not get the full speed of your CPU.
[E init_intrinsics_check.cc:43] CPU feature avx2 is present on your machine, but the Caffe2 binary is not compiled with it. It means you may not get the full speed of your CPU.
[E init_intrinsics_check.cc:43] CPU feature fma is present on your machine, but the Caffe2 binary is not compiled with it. It means you may not get the full speed of your CPU.
[INFO: train_net.py:  104]: Checkpoint directory created: /tmp/baseline-output/checkpoints
[WARNING: cnn.py:   25]: [====DEPRECATE WARNING====]: you are creating an object from CNNModelHelper class which will be deprecated soon. Please use ModelHelper object with brew module. For more information, please refer to caffe2.ai and python/brew.py, python/brew_test.py for more information.
[INFO: ava.py:   98]: Finished loading annotations from
[INFO: ava.py:  100]:   data/ava/annotations/ava_val_predicted_boxes.csv
[INFO: ava.py:  101]: Number of unique boxes: 23011
[INFO: ava.py:  102]: Number of annotations: 0
[INFO: ava.py:  123]: 12880 keyframes used.
[INFO: ava.py:  285]: === AVA dataset summary ===
[INFO: ava.py:  286]: Split: val
[INFO: ava.py:  287]: Use LFB? False
[INFO: ava.py:  288]: Detection threshold: 0.9
[INFO: ava.py:  290]: Full evaluation? False
[INFO: ava.py:  291]: Spatial shift position: 1
[INFO: ava.py:  292]: Number of videos: 64
data/ava/frames/1j20qq1JyX4/1j20qq1JyX4_000001.jpg
[INFO: ava.py:  296]: Number of frames: 1729931
[INFO: ava.py:  297]: Number of key frames: 12880
[INFO: ava.py:  298]: Number of boxes: 23011.
[INFO: ava_data_input.py:   51]: Creating the execution context for worker_ids: [100, 101, 102, 103], batch size: 16
[INFO: data_input_helper.py:  157]: CREATING EXECUTION CONTEXT
[INFO: data_input_helper.py:  164]: POOLS: {}
[INFO: data_input_helper.py:  165]: SHARED DATA LISTS: 0
[INFO: data_input_helper.py:  177]: worker_id: 100 list: 1
[INFO: data_input_helper.py:  179]: worker_id: 100 list keys: [100]
[INFO: data_input_helper.py:  177]: worker_id: 101 list: 2
[INFO: data_input_helper.py:  179]: worker_id: 101 list keys: [100, 101]
[INFO: data_input_helper.py:  177]: worker_id: 102 list: 3
[INFO: data_input_helper.py:  179]: worker_id: 102 list keys: [100, 101, 102]
[INFO: data_input_helper.py:  177]: worker_id: 103 list: 4
[INFO: data_input_helper.py:  179]: worker_id: 103 list keys: [100, 101, 102, 103]
[INFO: data_input_helper.py:  224]: CREATED POOL: {100: <multiprocessing.pool.Pool object at 0x7fc0d1c54ad0>, 101: <multiprocessing.pool.Pool object at 0x7fc0b7f8fa10>, 102: <multiprocessing.pool.Pool object at 0x7fc0b5321e90>, 103: <multiprocessing.pool.Pool object at 0x7fc0bdec6a90>}
[INFO: data_input_helper.py:  225]: CREATED LISTS: 4
[INFO: data_input_helper.py:  226]: POOL keys: [100, 101, 102, 103]
[INFO: data_input_helper.py:  227]: LIST keys: [100, 101, 102, 103]
[INFO: data_parallel_model.py:  174]: Parallelizing model for devices: [0]
[INFO: data_parallel_model.py:  197]: Create input and model training operators
[WARNING: data_parallel_model.py:  214]: 
[WARNING: data_parallel_model.py:  215]: ############# WARNING #############
[WARNING: data_parallel_model.py:  217]: Model CNN/<models.model_builder_video.ModelBuilder object at 0x7fc1810d6690> is used for testing/validation but
[WARNING: data_parallel_model.py:  218]: has init_params=True!
[WARNING: data_parallel_model.py:  219]: This can conflict with model training.
[WARNING: data_parallel_model.py:  220]: Please ensure model = ModelHelper(init_params=False)
[WARNING: data_parallel_model.py:  221]: ####################################
[WARNING: data_parallel_model.py:  222]: 
[INFO: data_parallel_model.py:  230]: Model for GPU : 0
[INFO: resnet_video.py:  150]: --------------- ResNet-50 1x64d-bottleneck_transformation_3d, ava, val, infer LFB? False, suffix: "_test" ---------------
[INFO: resnet_video.py:  166]: use_temp_convs_set: [[2], [1, 1, 1], [1, 0, 1, 0], [1, 0, 1, 0, 1, 0], [0, 1, 0]]
[INFO: resnet_video.py:  167]: temp_strides_set: [[1], [1, 1, 1], [1, 1, 1, 1], [1, 1, 1, 1, 1, 1], [1, 1, 1]]
[INFO: resnet_helper.py:   62]: res2_0 using dilation 1
[INFO: resnet_helper.py:   62]: res2_1 using dilation 1
[INFO: resnet_helper.py:   62]: res2_2 using dilation 1
[INFO: resnet_helper.py:   62]: res3_0 using dilation 1
[INFO: resnet_helper.py:   62]: res3_1 using dilation 1
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[INFO: resnet_helper.py:   62]: res3_2 using dilation 1
[INFO: resnet_helper.py:   62]: res3_3 using dilation 1
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[INFO: resnet_helper.py:   62]: res4_0 using dilation 1
[INFO: resnet_helper.py:   62]: res4_1 using dilation 1
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[INFO: resnet_helper.py:   62]: res4_2 using dilation 1
[INFO: resnet_helper.py:   62]: res4_3 using dilation 1
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[INFO: resnet_helper.py:   62]: res4_4 using dilation 1
[INFO: resnet_helper.py:   62]: res4_5 using dilation 1
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[INFO: resnet_helper.py:   62]: res5_0 using dilation 2
[INFO: resnet_helper.py:   62]: res5_1 using dilation 2
[INFO: resnet_helper.py:   62]: res5_2 using dilation 2
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: RoIAlign.
[WARNING: model_helper.py:  442]: You are creating an op that the ModelHelper does not recognize: Reshape.
[INFO: data_parallel_model.py:  275]: Parameter update function not defined --> only forward
*** Error in `python2': malloc(): memory corruption: 0x0000558c103b2920 ***

It seems the error occurs when running the following codes
`workspace.CreateNet(test_model.net)`
How can I solve this problem?

**Comment by leviswind** [comment]
After recompiling caffe2 with anaconda environment, it now has been solved.

**Comment by chaoyuaw** [comment]
Glad that you solved it. Please feel free to reopen if you have further questions.
